backends/ebpf/CMakeLists.txt

```diff
@@ -61,7 +61,20 @@ set (P4C_EBPF_SRCS
   ebpfModel.cpp
   midend.cpp
   lower.cpp
+  ../bmv2/common/JsonObjects.cpp
```

Comment:
After transferring the CodeGenerator code into the ProgramStructure class, some include statements are shifted to _portableProgramStructure.h_ file. I have added corresponding sources here. It is the exact reason for the changes in the _backends/tc/CMakeLists.txt_ file.

Reply:
> After transferring the CodeGenerator code into the ProgramStructure class, some include statements are shifted to portableProgramStructure.h file.

I don't see any code changes under the `ebpf` dir. Could you link to an example of relevant include statements?

Reply:
I didn't make any code changes under the `ebpf` directory. `ebpf` includes the `psa_switch` code.
 https://github.com/p4lang/p4c/blob/b27702f146526c5ebc8f04d330a9d0f5014f6205/backends/ebpf/ebpfBackend.cpp#L19
https://github.com/p4lang/p4c/blob/b27702f146526c5ebc8f04d330a9d0f5014f6205/backends/ebpf/psa/backend.cpp#L19

Now, Some of the `psa_switch` code depends on portable classes. For linking purposes while building the `ebpf` target, I have added portable sources in the `ebpf/CMakeLists.txt` file.

Previously, the `PsaCodeGenerator` class was in a separate file, and the above header files are needed for the `PsaCodeGenerator` class.
Ebpf didn't use codeGenerator classes. So, these header files are not in the `ebpf/CMakeLists.txt` file. 

I moved the `CodeGenerator` code into the ProgramStructure class and added these header sources to the `epbf/CMakeLists.txt` file.

Reply:
I'm not that familiar with CMake but I wonder if there is a better way to handle this kind of transitive dependencies in CMake?

What I see is that `ebpf` code is only directly dependent on `psaSwitch`. What has changed are the dependencies of `psaSwitch`. I imagine there could be a way to only update `psaSwitch` dependencies, and `ebpf` could know these transitive dependencies automatically without having to list them here explicitly.

I know how to do this in Bazel but not in CMake. Maybe @fruffy has some ideas?

Reply:
I did some searches on "CMake transitive dependency" and there does seem to be some solution. @rupesh-chiluka-marvell could you take a further look at this?

Reply:
Ok

Reply:
> I'm not that familiar with CMake but I wonder if there is a better way to handle this kind of transitive dependencies in CMake?

They are usually handled by making a library or include `PUBLIC`.

Reply:
This seems like a problem that should have been fixed earlier. eBPF should not include a BMv2 folder class. Let me take a look.

Reply:
The dependencies on BMv2 are completely messed up here. I will create a PR to clean this up before it gets even worse.

Reply:
#4770

Reply:
#4770 The PR is now merged, please rebase your changes on top of it.